core/src/main/scala/com/karumi/shot/screenshots/ScreenshotsSaver.scala

```diff
@@ -40,6 +48,35 @@ class ScreenshotsSaver {
     deleteFolder(destinyFolder)
   }
 
+  def copyOnlyFailingRecordedScreenshotsToTheReportFolder(
+      projectFolder: Folder,
+      flavor: String,
+      buildType: String,
+      destinyFolder: Folder,
+      updatedComparison: ScreenshotsComparisionResult): Unit = {
+
+    updatedComparison.errors.foreach {
+      case ScreenshotNotFound(screenshot) =>
+        copyFile(screenshot, destinyFolder)
+
+      case DifferentScreenshots(screenshot, base64Diff) =>
+        copyFile(screenshot, destinyFolder)
+
+      case DifferentImageDimensions(screenshot,
+                                    originalDimension,
+                                    newDimension) =>
+        copyFile(screenshot, destinyFolder)
+
+      case _ => // Nothing to do
+    }
```

Comment:
Instead of using this pattern matching where we use the same copy method in every scenario why don't we filter the errors we want to use first, and then we map the values to get the screenshot and use the copy? I think that would be easier to maintain in the future and also easier to read 😃

Reply:
True, it is more because I am not that deep into Scala and could not test it, so I better tried not do "fancy" things, which might break.
Is there a reason why it requires beta of Android Studio? This makes it hard to test for me.

Reply:
You can edit scala code from IntelliJ or any other IDE like visual studio code if you want. The Android Studio requirement is only needed for the shot-compose-consumer.

Reply:
I simplified it.